components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -535,10 +535,12 @@ open class GleanInternalAPI internal constructor () {
         // If any ping is being serialized to disk, wait for the to finish before spinning up
         // the WorkManager upload job.
         if (pingSerializationTasks.any()) {
-            // Await the serialization tasks. Once the serialization tasks have all completed,
-            // we can then safely enqueue the PingUploadWorker.
-            pingSerializationTasks.joinAll()
-            PingUploadWorker.enqueueWorker()
+            Dispatchers.API.launch {
+                // Await the serialization tasks. Once the serialization tasks have all completed,
+                // we can then safely enqueue the PingUploadWorker.
+                pingSerializationTasks.joinAll()
```

Comment:
I feel like this join could be removed given this refactor, but also vaguely feel like we tried that already...

Reply:
Oh never mind -- it's because the jobs being joined on are on the IO thread, and we join them in the single dispatcher thread.  Seems necessary.